lib/internal/modules/esm/resolve.js

```diff
@@ -76,107 +75,6 @@ function tryStatSync(path) {
   }
 }
 
-function readIfFile(path) {
-  let fd;
-  try {
-    fd = openSync(path, 'r');
-  } catch {
-    return undefined;
-  }
-  try {
-    if (!fstatSync(fd).isFile()) return undefined;
-    return readFileSync(fd, 'utf8');
-  } finally {
-    closeSync(fd);
-  }
-}
-
-function getPackageConfig(path, base) {
-  const existing = packageJSONCache.get(path);
-  if (existing !== undefined) {
-    if (!existing.isValid) {
-      throw new ERR_INVALID_PACKAGE_CONFIG(path, fileURLToPath(base), false);
```

Comment:
I'd have expected removing this would have caused a failure, but it seems we don't have a test for invalid JSON syntax in the package.json.

Can we please add one to this PR? Along the lines of -

test/fixtures/node_modules/invalid-pjson/package.json:
```js
{
  "invalid"
}
```

Where the test case then tries to load the package, which should throw an `ERR_INVALID_PACKAGE_CONFIG` error.

The CommonJS failure may be different here and I suppose that divergence is ok since CommonJS should throw as it historically has, but the ESM error is supposed to be the `ERR_INVALID_PACKAGE_CONFIG` by the current spec and this code path.